### PR TITLE
Rename link filter constructors per guidelines.

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
No functional/test changes.

### CODE REVIEW

File: link_options.go [hunk 1]
```diff
@@ -205,9 +206,10 @@ func LinkFilterSource(name string, code uint64, value interface{}) LinkFilter {
 	}
 }
 
-// LinkFilterSelector creates or updates the selector filter (apache.org:selector-filter:string) for this LinkFilter.
-func LinkFilterSelector(filter string) LinkFilter {
-	return LinkFilterSource(selectorFilter, selectorFilterCode, filter)
+// NewSelectorLinkFilter creates a new selector filter (apache.org:selector-filter:string) with the specified filter value.
+// Any preexisting selector filter will be updated with the new filter value.
+func NewSelectorLinkFilter(filter string) LinkFilter {
```
Comment (RickWinter) [inline link_options.go:211]: Should this be `NewLinkFilterSelector`?  That roles off the tongue better :), does it break a guideline?

**Review by RickWinter — APPROVED** [review]
Looks good, suggestion on the name in my review.

File: link_options.go (hunk 1 above)
Comment (jhendrixMSFT) [reply link_options.go:211]: Selector is a specialized instance of a link filter, see [here](https://github.com/Azure/go-amqp/blob/main/link_options.go#L177-L189).  Given this, I felt that `NewSelectorLinkFilter()` made more sense.

I thought about `NewSelectorFilter()`, but it returns a `LinkFilter` so I didn't go with that.

Comment (jhendrixMSFT) [reply link_options.go:211]: Hmm maybe `NewSelectorFilter()` isn't so bad.  Thoughts?

Comment (jhendrixMSFT) [reply link_options.go:211]: Changed to `NewSelectorFilter()`.

**Pull request merged**